packages/discord.js-utilities/src/lib/type-guards.ts

```diff
@@ -26,7 +26,7 @@ export function isCategoryChannel(channel: ChannelTypes | Nullish): channel is C
  * Checks whether a given channel is a {@link DMChannel}
  * @param channel The channel to check
  */
-export function isDMChannel(channel: ChannelTypes | Nullish): channel is DMChannel {
+export function isDMChannel(channel: ChannelTypes | Nullish): channel is DMChannel | PartialDMChannel {
```

Comment:
In what instance does a bot get a PartialDMChannel?

Regardless, I think we should ensure this always returns a DMChannel by doing something like `return channel?.type === 'DM' && whateverCheckThatIsFalseForPartial`

Reply:
> In what instance does a bot get a PartialDMChannel?

When you use the `CHANNELS` partial and you get a DM message.